File: suwan/Programmers/step1/시저 암호.py
print(ord('A'))
print(ord('Z'))
print(ord('a'))
print(ord('z'))


def solution(s, n):
    answer = ''
    for a in s:
        a_ord = ord(a)
        if 65 <= a_ord <= 90:
            a_ord += n
            if a_ord > 90:
                a_ord -= 26
        if 97 <= a_ord <= 122:
            a_ord += n
            if a_ord > 122:
                a_ord -= 26
        answer += chr(a_ord)

    return answer


s1 = "AB"
n1 = 1  # return "BC"

s2 = "z"
n2 = 1  # return "a"

s3 = "a B z"
n3 = 4  # return "e F d"

print(solution(s1, n1))
print(solution(s2, n2))
print(solution(s3, n3))

File: mmkser10/javascript/programmers/시저_암호.js
function solution(s, n) {
    var answer = '';
    for(let i = 0 ; i < s.length ; i++){
        if("a".charCodeAt(0) <= s.charCodeAt(i) && "z".charCodeAt(0) >= s.charCodeAt(i)){
            answer = answer + String.fromCharCode(((s.charCodeAt(i)+n)%"a".charCodeAt(0))%26 + "a".charCodeAt(0) );
        }
        else if("A".charCodeAt(0) <= s.charCodeAt(i) && "Z".charCodeAt(0) >= s.charCodeAt(i)){
            answer = answer + String.fromCharCode(((s.charCodeAt(i)+n)%"A".charCodeAt(0))%26 + "A".charCodeAt(0) );
        }
        else
            answer = answer + s.charAt(i);
    }
    return answer;
}

// another sol
function anotherSolutionA(s, n){
    let upper = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    let lower = "abcdefghijklmnopqrstuvwxyz";
    let answer= '';

    for(let i = 0 ; i < s.length ; i++){
        let text = s[i];
        if(text == ' ') {
            answer += ' ';
            continue;
        }
        let textArr = upper.includes(text) ? upper : lower;
        let index = textArr.indexOf(text)+n;
        if(index >= textArr.length) index -= textArr.length;
        answer += textArr[index];
    }
    return answer;
}